src/TorchSharp/Tensor/Tensor.Factories.cs

```diff
@@ -1381,11 +1389,11 @@ public static Tensor tensor(bool[] dataArray, long[] dimensions, torch.ScalarTyp
                     deleters.TryRemove(deleter, out deleter!);
                 });
```

Comment:
This is a little confusing form me. it takes IntPtr as a parameter and not used while it uses `gchp` which is defined outside the delegate. does the caller of this delegate pass the right value to use inside? 

Note, using `gchp` it will force allocating a closure object wrapping `gchp` to use when the delegate get called.

Reply:
This really old code that has survived since before I started working on TorchSharp, so I don't actually know the exact details. The basic idea is that the underlying storage is shared, so there's no copy, and once the tensor is dead, the pinning of the .NET array is no longer needed. If there's a better way to accomplish that...

Reply:
The delegate will be called back from the native runtime when the tensor goes away.

Reply:
and the native caller will pass the value to the delegate which be exact to `gchp` in this case. right?

Reply:
Just ran it in the debugger -- ptr and ghcp are different when the delegate is called, so it seems we need to capture ghcp.

Reply:
ok, but this makes me worry more as the native code asks to delete something else. I hope we are not leaking anything here. It would be nice if someone could investigate later to understand how things work here. This is not a blocker to this PR as we already had this logic before your change.

Reply:
Could be the native code is calling with the address `dataArrayAddr`. It is better to keep as it is for now.

Reply:
I'll open an issue on this -- investigate why ghcp and ptr are different.

Reply:
https://github.com/dotnet/TorchSharp/issues/672